app/src/main/java/com/keylesspalace/tusky/components/filters/EditFilterActivity.kt

```diff
@@ -256,6 +262,32 @@ class EditFilterActivity : BaseActivity() {
         }
     }
 
+    private fun deleteFilter() {
+        originalFilter?.let { filter ->
```

Comment:
Can this be flattened a bit?

5 or 6 levels seem a bit much.

(A more linear call graph maybe? No fold for example? Could also reduce the error case to one.)